modules/tow-boot/uswid.nix

```diff
@@ -0,0 +1,128 @@
+{ config, lib, pkgs, ... }:
+
+let
+  inherit (lib)
+    concatStringsSep
+    mkOption
+    optional
+    types
+  ;
+
+  inherit (config.Tow-Boot)
+    outputName
+    towBootIdentifier
+    uBootVersion
+    variant
+  ;
+
+  inherit (config)
+    device
+  ;
+
+  #
+  # For the definition of the fields, refer to:
+  #
+  #  - ISO/IEC 19770-2:2015 (Preferred)
+  #      - https://standards.iso.org/iso/19770/-2/2015/schema.xsd
+  #      - https://web.archive.org/web/20230824211322/https://standards.iso.org/iso/19770/-2/2015/schema.xsd
+  #
+  #  - RFC9393 (Adequate)
+  #      - https://datatracker.ietf.org/doc/html/rfc9393
+  #
+  #  - NIST.IR 8060 (Okay, too)
+  #      - https://nvlpubs.nist.gov/nistpubs/ir/2016/NIST.IR.8060.pdf
+  #
+  uSWIDData = {
+    "uSWID" = {
+      # Will be replaced during execution with basename of $out (guaranteed meaningfully unique)
+      # DO NOT assume any useful form when consuming the SWID.
+      tag-id = "$id";
+
+      # (Unchanging) meaningful name for end-user
+      software-name = "${outputName} for ${device.name} [variant: ${variant}]";
+
+      # Short description for the software (shown in e.g. fwupd frontends)
+      summary = "An opinionated distribution of U-Boot";
+
+      # Lexicographically sortable version name
+      # (pre/post suffixes will not sort as expected, but that is okay)
+      software-version = "${"${uBootVersion}-${towBootIdentifier}"}";
```

Comment:
This currently looks like:

```
2023.07-007-rc2-pre
```

Actual releases will look like:

```
2023.07-007
```

There is no equivalent to a "patch" release of Tow-Boot... If a release needs to happen quickly, our release number (007 here) is increased. The prefix will always increase, and follows the upstream (U-Boot) versioning scheme.